### DISCUSSION TITLE
Skip validation for the first few epochs

### DISCUSSION THREAD

**Discussion by sangrockEG** [Lightning-AI/pytorch-lightning#12873] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi, PL.

Is there any way to skip validation for the first few epochs (ex. 10)?
I searched for an hour, but the only thing I found is check_val_every_n_epoch.

But the thing I want is different from this.
I want to check every epoch after a certain epoch.

Since I'm a beginner, any comment would be greatly appreciated.

**Comment by akihironitta** [comment, marked as answer]
Hi @sangrockEG! You can utilise `self.current_epoch` in your LightningModule:

```python
def validation_step(self, batch, batch_idx):
    if self.current_epoch <= 9:
        return
```
https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#current-epoch

**Reply by sangrockEG** [reply]
Hi @akihironitta!
Thanks for your response, but I already tried that solution.
I think I should clarify my problem, sorry about it.

Since my validation loader includes a time-consuming process, I want to completely skip it.
I do not want to enter the validation loop at all.
(And of course, skipping in the validation_step method can't avoid it.)

Anyway, I solved my problem with save and load.
At first, train with check_val_every_n_epoch=10, and then stop it when the epoch reaches 10.
And then load the checkpoint, re-start with check_val_every_n_epoch=1.
It works but is not fancy at all, you know...

**Reply by akihironitta** [reply]
@sangrockEG Thanks for your clarification. In that case, maybe you could also try changing its value somewhere in your `LightningModule` (or in a `Callback`). I believe the following code snippet should work:
```python
trainer = Trainer(..., check_val_every_n_epoch=999999)

class YourLightningModule(LightningModule):
    def on_train_epoch_end(self):
        if self.current_epoch == 10:
            self.trainer.check_val_every_n_epoch = 1
```

**Reply by sangrockEG** [reply]
@akihironitta 
Thanks a lot for your solution!
